glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+class HttpPingUploader {
+    let LOG_TAG = "glean/HttpPingUploader"
+
+    func logPing(path: String, data: String, config: Configuration) {
+        if config.logPings {
+            NSLog("\(LOG_TAG) : \(path)\n\(data)")
+        }
+    }
+
+    func upload(path: String, data: String, config: Configuration, callback: @escaping (Bool, Error?) -> Void) {
```

Comment:
What's the `@escaping` part for?

Reply:
Oh one thing I'd like to learn about Swift (and this is as good a place as anywhere else):

Are arguments always passed by value? Or does Swift auto-reference if needed? Or can you explicitly pass references?

Reply:
The `@escaping` marks this closure as escaping. If a closure is passed as an argument to a function and it is invoked after the function returns, the closure is escaping. You can also think of it like the closure argument escapes the function body.  A while back Swift changed the default to noescaping and removed the `@noescaping` so we now have to explicitly mark escaping closures with `@escaping` so the compiler knows you are aware of the implications.

Reply:
> Oh one thing I'd like to learn about Swift (and this is as good a place as anywhere else):
> 
> Are arguments always passed by value? Or does Swift auto-reference if needed? Or can you explicitly pass references?

Most cases, things are passed by value, including `struct` and `enum`. But **not** `class`.  Class is always treated as a reference type in Swift and is passed by reference.  If you want to force the pass by reference to a value type, you can use the `inout` keyword in the parameter list.